sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfigurationprovider/_azureappconfigurationprovider.py

```diff
@@ -0,0 +1,218 @@
+
+        provider._trim_prefixes = sorted(kwargs.pop(
+            "trimmed_key_prefixes", []), key=len, reverse=True)
+
+        provider._dict = {}
+
+        secret_clients = key_vault_options.secret_clients if key_vault_options else {}
+
+        for select in selects:
+            try:
+                configurations = provider._client.list_configuration_settings(
+                    key_filter=select.key_filter, label_filter=select.label_filter)
+            except HttpResponseError as e:
+                raise e
+            for config in configurations:
+                if config.content_type == KEY_VAULT_REFERENCE_CONTENT_TYPE:
+                    provider.__resolve_keyvault_references(
+                        config, key_vault_options, secret_clients)
+                elif provider.__is_json_content_type(config.content_type):
+                    try:
+                        j_object = json.loads(config.value)
+                        provider._dict[provider.__trim(config.key)] = j_object
+                    except json.JSONDecodeError as e:
+                        raise ValueError(
+                            "Invalid JSON value for key: " + config.key)
+                else:
+                    provider._dict[provider.__trim(config.key)] = config.value
+        return provider
+
+    def __buildprovider(self, connection_string, endpoint, credential, key_vault_options):
+        headers = {}
+        correlation_context = "RequestType=Startup"
+
+        if (key_vault_options and
+                (key_vault_options.credential or key_vault_options or key_vault_options.secret_resolver)):
+            correlation_context += ",UsesKeyVault"
+
+        headers["Correlation-Context"] = correlation_context
+        useragent = USER_AGENT
+
+        if (connection_string and endpoint):
+            raise AttributeError(
+                "Both connection_string and endpoint are set. Only one of these should be set.")
+
+        if connection_string:
+            self._client = AzureAppConfigurationClient.from_connection_string(
+                connection_string, user_agent=useragent, headers=headers)
+            return
+        self._client = AzureAppConfigurationClient(
+            endpoint, credential, user_agent=useragent, headers=headers)
+
+    def __resolve_keyvault_references(self, config, key_vault_options, secret_clients):
+        if key_vault_options is None:
+            raise AttributeError(
+                "Key Vault options must be set to resolve Key Vault references.")
+
+        if config.secret_id is None:
+            raise AttributeError(
+                "Key Vault reference must have a uri value.")
+
+        key_vault_identifier = KeyVaultSecretIdentifier(config.secret_id)
+
+        referenced_client = None
+
+        for secret_client in secret_clients:
+            if secret_client.vault_url == key_vault_identifier.vault_url:
+                referenced_client = secret_client
+                break
```

Comment:
Python has a built-in [next](https://docs.python.org/3.8/library/functions.html#next) function for making this for loop simpler.